llama-index-core/llama_index/core/tools/function_tool.py

```diff
@@ -21,6 +21,16 @@ async def _async_wrapped_fn(*args: Any, **kwargs: Any) -> Any:
     return _async_wrapped_fn
 
 
+def async_to_sync(func_async: AsyncCallable) -> Callable:
+    """Async from sync."""
+
+    def _sync_wrapped_fn(*args: Any, **kwargs: Any) -> Any:
+        loop = asyncio.get_event_loop()
```

Comment:
IDK if we handle this anywhere else (to ensure there's a running loop), but this will raise a `RuntimeError` if there's no running event loop.

Reply:
Yea, switched to using our `asyncio_run()` util